src/runner/bootstrapper.js

```diff
@@ -21,48 +22,67 @@ export default class Bootstrapper {
         this.reporter = null;
     }
 
-    static async _convertAliasOrPathToBrowserInfo (browser, installations) {
-        if (typeof browser === 'string') {
-            if (browser in installations)
-                return installations[browser];
+    static async _convertPathAndCmdToBrowser (browserArg) {
+        var pathProvider = new BrowserProvider('path');
+        var options      = browserArg.cmd ? { cmd: browserArg.cmd } : null;
 
-            var pathDescriptorMatch = BROWSER_PATH_DESCRIPTOR_RE.exec(browser);
+        return new Browser(pathProvider, browserArg.path, options);
+    }
+
+    static async _convertAliasToBrowser (browserArg) {
+        var providerRegExpMatch = BROWSER_PROVIDER_RE.exec(browserArg);
+
+        if (!providerRegExpMatch)
+            throw new GeneralError(MESSAGE.cantFindBrowser, browserArg);
 
-            if (!pathDescriptorMatch)
-                throw new GeneralError(MESSAGE.cantFindBrowser, browser);
+        var providerName = providerRegExpMatch[1] || 'local';
+        var alias        = providerRegExpMatch[2];
 
-            var path        = pathDescriptorMatch[1];
-            var browserInfo = await getBrowserInfo(path);
+        var provider = new BrowserProvider(providerName);
```

Comment:
I guess it doesn't make sense to create browser provider each time we need it. It is something that should have one instance per testcafe instance. I think we need `browser-providers-pool` which will a)store built-in providers, b)maintain pool of available providers c)maybe move alias parsing logic there as well d)will have `getAvaliableAliases` method which will return aggregated results